frontend/components/product-list/sections/FilterableProductsSection/ProductListFilters/FilterRow.tsx

```diff
@@ -266,13 +245,74 @@ const Row = React.memo(
    })
 );
 
-function parseRange(value: string): [number, number] {
-   const [min, max] = value.split(':');
-   return [parseFloat(min), parseFloat(max)];
+interface PriceFilterProps {
+   priceRangeFacet: Facet;
+   showAllFacetValues: boolean;
+   onResize(): void;
+}
+
+function PriceFilter({
+   priceRangeFacet,
+   showAllFacetValues,
+   onResize,
+}: PriceFilterProps) {
+   const priceFacet = useFacet('price');
+   return (
+      <>
+         <ListFilter
+            key={priceRangeFacet.handle}
+            facet={priceRangeFacet}
+            multiple
+            dependentFacets={[priceFacet.handle]}
+            showAllValues={showAllFacetValues}
+            sortItems={sortByPriceRange}
+         />
+         <RangeFilterInput
+            facet={priceFacet}
+            minFieldPrefix="$"
+            minFieldPlaceholder="Min"
+            maxFieldPrefix="$"
+            maxFieldPlaceholder="Max"
+            dependentFacets={[priceRangeFacet.handle]}
+            onError={onResize}
+            onDismissError={onResize}
+         />
+      </>
+   );
+}
+
+const priceRangeRegex = /[^\d]*(\d+)\s*-\s*[^\d]*(\d+)/i;
+const priceLowerboundRegex = /[^\d]*(\d+)\+/i;
+
+function parseRange(value: string): [number | null, number | null] {
+   const [, min, max] = value.match(priceRangeRegex) ?? [];
+   if (typeof min === 'string' && typeof max === 'string') {
+      return [parseFloat(min), parseFloat(max)];
+   }
+   const [, lowerbound] = value.match(priceLowerboundRegex) ?? [];
+   if (typeof lowerbound === 'string') {
+      return [parseFloat(lowerbound), null];
+   }
+   return [null, null];
 }
 
 function sortByPriceRange(a: FacetOption, b: FacetOption): number {
-   const [aMin] = parseRange(a.value);
-   const [bMin] = parseRange(b.value);
+   const [aMin, aMax] = parseRange(a.value);
+   const [bMin, bMax] = parseRange(b.value);
+   if (aMax == null && aMin != null) {
+      return 1;
+   }
+   if (bMax == null && bMin != null) {
+      return -1;
+   }
+   if (aMin === bMin) {
+      return 0;
+   }
+   if (aMin == null) {
+      return 1;
+   }
+   if (bMin == null) {
+      return -1;
+   }
```

Comment:
Kind of a nit-pick here.. but I feel like this could be simplified as:

* extract all numbers from the range text
* find their avg
* compare that

```javsacript
const avg = (x) => {
   const nums = x.match(/\d+/g);
   return nums.reduce((x, y) => x + parseFloat(y), 0) / nums.length
};
return avg(a.value) - avg(a.value);
```

Reply:
Neat! 👏  I've refactored the code 👍